addons/sourcemod/scripting/ttt/ttt_inventory.sp

```diff
@@ -0,0 +1,690 @@
+	HookConVarChange(g_cv_CreditLootMax, OnConvarChanged);
+	HookConVarChange(g_cv_ItemLootMax, OnConvarChanged);
+	HookConVarChange(g_cv_CreditLootChance, OnConvarChanged);
+	HookConVarChange(g_cv_ItemLootChance, OnConvarChanged);
+	HookConVarChange(g_cv_LootChanceFallOff, OnConvarChanged);
+	
+	HookEvent("player_death", OnPlayerDeath_Pre, EventHookMode_Pre);
+}
+
+public void OnConfigsExecuted()
+{
+	g_bCreditLoot = GetConVarBool(g_cv_EnableCreditLoot);
+	g_bItemLoot = GetConVarBool(g_cv_EnableItemLoot);
+	g_iCreditLootMax = GetConVarInt(g_cv_CreditLootMax);
+	g_iItemLootMax = GetConVarInt(g_cv_ItemLootMax);
+	g_fCreditLootChance = GetConVarFloat(g_cv_CreditLootChance);
+	g_fItemLootChance = GetConVarFloat(g_cv_ItemLootChance);
+	g_fLootChanceFallOff = GetConVarFloat(g_cv_LootChanceFallOff);
+	
+	g_cv_PluginTag = FindConVar("ttt_plugin_tag");
+	g_cv_PluginTag.AddChangeHook(OnConvarChanged);
+	g_cv_PluginTag.GetString(g_szPluginTag, sizeof(g_szPluginTag));
+}
+
+public Action Command_Inventory(int client, int args)
+{
+	if (!TTT_IsClientValid(client))
+	{
+		return Plugin_Handled;
+	}
+	
+	if (!TTT_IsRoundActive())
+	{
+		ReplyToCommand(client, "%T", "RoundInactive");
+		return Plugin_Handled;
+	}
+	
+	Menu inventoryMenu = CreateMenu(Menu_InventoryHandler);
+	StringMap playerInv = GetPlayerInventory(client);
+	if (playerInv == null)
+	{
+		ReplyToCommand(client, "You have no items registered in your inventory.");
+		return Plugin_Handled;
+	}
+	
+	
+	StringMapSnapshot inventoryListing = playerInv.Snapshot();
+	for (int i = 0; i < inventoryListing.Length; i++)
+	{
+		char itemShort[16];
+		inventoryListing.GetKey(i, itemShort, sizeof(itemShort));
+		
+		int amount;
+		GetTrieValue(inventoryListing, itemShort, amount);
+		
+		char itemLong[32];
+		if (TTT_GetItemLong(itemShort, itemLong, sizeof(itemLong)))
+		{
+			char display[64];
+			char szAmount[4];
+			
+			IntToString(amount, szAmount, sizeof(szAmount));
+			Format(display, sizeof(display), "%s (x%s)", itemLong, szAmount);
+			inventoryMenu.AddItem(itemShort, display);
+		}
+	}
+	inventoryMenu.SetTitle("%T", "InventoryTitle");
+	inventoryMenu.Display(client, MENU_TIME_FOREVER);
+	
+	CloseHandle(inventoryListing);
+	
+	return Plugin_Handled;
+}
+
+public int Menu_InventoryHandler(Menu menu, MenuAction action, int client, int itemNum)
+{
+	switch (action)
+	{
+		case MenuAction_Select:
+		{
+			if (!IsPlayerAlive(client))
+			{
+				CPrintToChat(client, "%s %T", g_szPluginTag, "YouAreDead", client);
+				return;
+			}
+			
+			char info[16];
+			GetMenuItem(menu, itemNum, info, sizeof(info));
+			
+			Call_StartForward(g_hOnInventoryMenuItemSelect);
+			Call_PushCell(client);
+			Call_PushString(info);
+			Call_Finish();
+		}
+		
+		case MenuAction_End:
+		{
+			CloseHandle(menu);
+		}
+	}
+}
+
+public void TTT_OnRoundStart(int innocents, int traitors, int detective)
+{
+	//Reset Inventories
+	StringMapSnapshot inventories = g_PlayerInventory.Snapshot();
```

Comment:
How about g_PlayerInventory.Clear?
http://sourcemod.net/new-api/adt_trie/StringMap/Clear

Reply:
Would it close the handles inside?

Reply:
Maybe between inventory and client loop. Otherwise you spam the stringmap with unused userids (when player left the server).